src/tls13.c

```diff
@@ -7190,13 +7190,8 @@ int DoTls13HandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                                         ENCRYPT_AND_DECRYPT_SIDE, 1)) != 0) {
                     return ret;
                 }
-        #ifdef WOLFSSL_EARLY_DATA
```

Comment:
The Encrypt side is set In SendTls13EndOfEarlyData() after building message.
This is after all the early data has been sent and is the last message that is encrypted with the EarlyData keys.
Setting the Encrypt side early should stop things working.

Checking ssl->earlyData == no_early_data and do both sides is probably the fix.